docs/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,196 @@
+use the email address of `terraform_service_account` that is created by the Terraform Example Foundation 0-bootstrap step.
+
+**Notes:**
+
+- If you see other quota errors, see the [Quota documentation](https://cloud.google.com/docs/quota).
+
+#### Terraform State Snapshot lock
+
+**Error message:**
+
+When running the build for the branch `production` in step 3-networks in your **Foundation Pipeline** the build fails with:
+
+```
+state snapshot was created by Terraform v0.x.x, which is newer than current v0.13.6; upgrade to Terraform v0.x.x or greater to work with this state
+```
+
+**Cause:**
+
+The manual deploy step for the shared environment in [3-networks](../3-networks#deploying-with-cloud-build) was execute with a Terraform version newer than version v0.13.6 used in the **Foundation Pipeline**.
+
+**Solution:**
+
+You will need to re-run the deploy of the 3-networks shared environment with Terraform v0.13.6
+
+Steps:
+
+- Go to folder `gcp-networks/envs/shared/`
+- Update `backend.tf` with your bucket name from the 0-bootstrap step
+- Run `terraform destroy` in the folder using the Terraform v0.x.x version.
+- Delete the Terraform state file in `gs://YOUR-TF-STATE-BUCKET/terraform/networks/envs/shared/default.tfstate`. This bucket is in your **Seed Project**.
+- Re-run the manual deploy of 3-networks shared environment using Terraform v0.13.6
+
+
+#### Application authenticated using end user credentials
+
+**Error message:**
+
+When running gcloud commands in Cloud Shell like
+
+```
+gcloud scc notifications describe <scc_notification_name> --organization YOUR_ORGANIZATION_ID
+```
+
+or
+
+```
+gcloud access-context-manager policies list --organization YOUR_ORGANIZATION_ID --format="value(name)"
+```
+
+you receive the error:
+
+```
+Error 403: Your application has authenticated using end user credentials from the Google Cloud SDK or Google Cloud Shell which are not supported by the X.googleapis.com.
+We recommend configuring the billing/quota_project setting in gcloud or using a service account through the auth/impersonate_service_account setting.
+For more information about service accounts and how to use them in your application, see https://cloud.google.com/docs/authentication/.
+```
+
+**Cause:**
+
+When using application default credentials in Cloud Shell a billing project is not available for APIs like `securitycenter.googleapis.com` or `accesscontextmanager.googleapis.com`.
+
+**Solution:**
+
+you can re-run the command using impersonation or providing a billing project:
+
+- Impersonate the Terraform Service Account
+
+```
+--impersonate-service-account=org-terraform@<SEED_PROJECT_ID>.iam.gserviceaccount.com
+```
+
+- Provide a billing project
+
+```
+--billing-project=<A-VALID-PROJECT-ID>
+```
+
+If you provide a billing project, you must have the `serviceusage.services.use` permission on the billing_project.
+
+#### Cannot assign requested address error in Cloud Shell
```

Comment:
lets make all these h3 for visibility with subheadings being h4?

Reply:
@bharathkkb 

You mean like this
```
# Troubleshooting
## Terminology
## Problems
## Common issues <- from h3 to h2
### Project quota exceeded <- from h4 to h3
### Terraform State Snapshot lock <- from h4 to h3
### Application authenticated using end user credentials <- from h4 to h3
### Cannot assign requested address error in Cloud Shell <- from h4 to h3
### Caller does not have permission in the Organization
### Billing quota exceeded
```
# Troubleshooting
## Terminology
## Problems
## Common issues
### Project quota exceeded
###  Terraform State Snapshot lock
### Application authenticated using end user credentials
### Cannot assign requested address error in Cloud Shell
### Caller does not have permission in the Organization
### Billing quota exceeded

Reply:
yup looks great now